server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java

```diff
@@ -490,40 +492,40 @@ public void resetMounting(boolean reverseYaw) {
 		mountRotFix.set(buffer);
 
 		// Get the difference from the last adjustment
-		buffer.multLocal(lastRotAdjust.inverseLocal());
+		buffer.multLocal(lastRotAdjust.inverse());
 		// Apply the yaw rotation difference to the yaw fix quaternion
-		yawFix.multLocal(buffer.inverseLocal());
+		yawFix.multLocal(buffer.inverse());
 	}
 
-	private void fixYaw(Quaternion reference) {
+	private void fixYaw(Quaternion sensorRotation, Quaternion reference) {
 		// Use only yaw HMD rotation
-		Quaternion targetRotation = reference.clone();
-		targetRotation.fromAngles(0, targetRotation.getYaw(), 0);
+		reference = reference.clone();
+		reference.fromAngles(0, reference.getYaw(), 0);
 
-		Quaternion sensorRotation = getMountedAdjustedRotation();
+		sensorRotation = sensorRotation.clone();
 		gyroFix.mult(sensorRotation, sensorRotation);
 		sensorRotation.multLocal(attachmentFix);
 		sensorRotation.multLocal(mountRotFix);
 
 		sensorRotation.fromAngles(0, sensorRotation.getYaw(), 0);
 
-		yawFix.set(sensorRotation).inverseLocal().multLocal(targetRotation);
+		yawFix.set(sensorRotation.inverse().mult(reference));
 	}
 
 	private void calibrateMag() {
 		if (magCalibrationStatus >= CalibrationAccuracy.HIGH.status) {
 			magnetometerCalibrated = true;
 			// During calibration set correction to match magnetometer readings
 			// TODO : Correct only yaw
-			correction.set(rotQuaternion).inverseLocal().multLocal(rotMagQuaternion);
+			correction.set(rotQuaternion.inverse().mult(rotMagQuaternion));
```

Comment:
It's specifically ordered to first set, then inverseLocal and multLocal so nothing outside is modified and clonning is not needed.